lib/pwsh/util.rb

```diff
@@ -48,9 +48,13 @@ def snake_case(string)
     # Iterate through a hashes keys, snake_casing them
     #
     # @return [Hash] Hash with all keys snake_cased
-    def snake_case_hash_keys(hash)
+    def snake_case_hash_keys(object)
+      return object.map { |h| snake_case_hash_keys(h) } if object.is_a?(Array)
+      return object unless object.is_a?(Hash)
+
       modified_hash = {}
-      hash.each do |key, value|
+      object.each do |key, value|
+        value = value.map { |h| snake_case_hash_keys(h) } if value.is_a?(Array)
         value = snake_case_hash_keys(value) if value.is_a?(Hash)
```

Comment:
`value = snake_case_hash_keys(value)` would be a much easier version of those two lines. Same below in `pascal_case_hash_keys`' hash loop.